src/pilot/assign_role.py

```diff
@@ -911,17 +911,16 @@ def generate_osd_config(ip_mac_service_tag, drac_client):
     new_osd_config = None
     if len(ssds) > 0 and len(spinners) == 0:
         # If we have an all flash config, then let Ceph colocate the journals
-        new_osd_config = generate_osd_config_without_journals(controllers,
-                                                              ssds)
+        new_osd_config, mklvm = generate_osd_config_without_journals(controllers,
+                                                              ssds, system_id)
     elif len(ssds) > 0 and len(spinners) > 0:
         # If we have a mix of flash and spinners, then use the ssds as journals
-        new_osd_config = generate_osd_config_with_journals(controllers,
-                                                           spinners,
-                                                           ssds)
+        new_osd_config, mklvm = generate_osd_config_with_journals(controllers,
```

Comment:
I did not know you could do this... Does the variable assignments receive the same value?

Reply:
Yes you can, the methods now returns osd_config, mklvm